azure/aks/modules/network/main.tf

```diff
@@ -0,0 +1,83 @@
+resource "azurerm_virtual_network" "aks_vnet" {
+  name                = "${var.resource_prefix}-vnet"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  address_space       = ["10.1.0.0/16"]
```

Comment:
do I understand it correctly that the vnet address_space is for the whole network and must include whatever you assign to your subnets?
Besides that it's dynamically used for assigning pod IPs etc.

While the `aks_subnet` is solely for VM purposes? So I could run a max of 256 nodes?
Same for the DBs I assume

Reply:
I see the cluster having a different range.

```
Pod CIDR
10.244.0.0/16
Service CIDR
10.0.0.0/16
```

How does it relate to the vnet?

Probably an architecture diagram could help to get an overview.